src/java/com/puppetlabs/puppetserver/ShellUtils.java

```diff
@@ -66,9 +66,7 @@ private static ExecutionResult executeExecutor(CommandLine commandLine,
      */
     public static ExecutionResult executeCommand(String command)
             throws InterruptedException, IOException {
-        CommandLine commandLine = CommandLine.parse(command);
```

Comment:
Sorry, they aren't always a single string. They are either a command as a string, or an array where the command is the first element and the arguments are the rest of the array elements (could be one string or one array element per argument).